packages/devtools_app/lib/src/screens/performance/rebuild_counts.dart

```diff
@@ -20,112 +21,297 @@ import '../../primitives/utils.dart';
 //       "names": ["PlanetsApp", "MaterialApp", ...]
 //     }
 //   },
-//   "newLocations": {
-//     "file": [id1, line1, column1, id2, line2, column2, ...],
-//     }
-//   }
 // }
 
-class RebuildCountModel {
-  final Map<int, RebuildLocation> _locationMap = <int, RebuildLocation>{};
-
+class _RebuildCountStats {
+  ValueListenable<List<RebuildLocation>> get locations => _locations;
   final _locations = ListValueNotifier<RebuildLocation>(<RebuildLocation>[]);
 
-  ValueListenable<List<RebuildLocation>> get locations => _locations;
+  final _rebuildLocationMap = <int, RebuildLocation>{};
 
-  void processRebuildEvent(Map<String, dynamic> json) {
-    // parse locations
-    if (json.containsKey('locations')) {
-      final fileLocationsMap =
-          (json['locations'] as Map).cast<String, dynamic>();
-
-      for (final String file in fileLocationsMap.keys) {
-        final entries =
-            (fileLocationsMap[file] as Map).cast<String, List<dynamic>>();
-
-        final ids = entries['ids']!.cast<int>();
-        final lines = entries['lines']!.cast<int>();
-        final columns = entries['columns']!.cast<int>();
-        final names = entries['names']!.cast<String>();
-
-        for (var i = 0; i < ids.length; i++) {
-          final location = RebuildLocation(
-            id: ids[i],
-            path: file,
+  void incrementCount(Location location, int count) {
+    final id = location.id;
+    final existing = _rebuildLocationMap[id];
+    if (existing != null) {
+      existing.buildCount += count;
+    } else {
+      final rebuildLocation = RebuildLocation(location, buildCount: count);
+      _locations.add(rebuildLocation);
+      _rebuildLocationMap[id] = rebuildLocation;
+    }
+  }
+
+  void clear() {
+    _locations.clear();
+    _rebuildLocationMap.clear();
+  }
+
+  void dataChanged() {
+    _locations.notifyListeners();
+  }
+}
+
+/// Mapping from ids to [Location] objects.
+///
+/// Some Locations may be unresolved with ids but no valid path while valid
+/// paths are being fetched from the running Flutter application.
+class LocationMap {
+  LocationMap();
+
+  LocationMap.fromJson(Map<String, dynamic> json) {
+    processLocationMap(json);
+  }
```

Comment:
can this be a factory constructor or a static method?

Reply:
what is the advantage of using a factory constructor? this is a little more terse for this case.